src/catch2/internal/catch_random_number_generator.cpp

```diff
@@ -7,6 +7,12 @@
 // SPDX-License-Identifier: BSL-1.0
 #include <catch2/internal/catch_random_number_generator.hpp>
 
+#if defined( __clang__ )
+#    define CATCH2_CLANG_NO_SANITIZE_INTEGER \
+        __attribute__( ( no_sanitize( "undefined", "integer" ) ) )
```

Comment:
Thanks, but reading the [documentation here](https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html#disabling-instrumentation-with-attribute-no-sanitize-undefined), I think this is supposed to be 

```suggestion
        __attribute__( ( no_sanitize( "unsigned-integer-overflow" ) ) )
```

instead, to only target unsigned integer overflows.